PixelHunter1995/SceneLib/SceneParser.cs

```diff
@@ -103,12 +104,63 @@ public static Scene ParseSceneXml(string sceneXmlPath)
                     drawables.Add(walkingArea);
 
                 }
+                else if (node.Name == "objectgroup" && node.Attributes["name"]?.InnerText == "portals")
+                {
+                    foreach (XmlNode portalNode in node.ChildNodes)
+                    {
+                        Portal portal = ParsePortalNode(portalNode, sceneWidth);
+                        portals[portal.Name] = portal;
```

Comment:
It would make a bit more sense to me if the Scene took care of making dictionaries out of them. But it's not really important.